File: cpp/singly-llist.cpp
// #include <bits/stdc++.h>
#include <iostream>
using namespace std;

class Node
{
public:
    int data;
    Node *next;
};

int main()
{
    Node *head;
    Node *one = NULL;
    Node *two = NULL;
    Node *three = NULL;

    // Allocate 3 nodes in the heap
    one = new Node();
    two = new Node();
    three = new Node();

    // Assign values to data
    one->data = 1;
    two->data = 2;
    three->data = 3;

    // Connect the nodes
    one->next = two;
    two->next = three;
    three->next = NULL;

    // Print the Linked list value
    head = one;
    while (head != NULL)
    {
        cout << head->data;
        head = head->next;
    }
}

File: java/singly-llist.java
// package DataStructures.Java;
// public class singly-llist {
// }asd

class LinkedList {
    Node head;

    static class Node {
        int data;
        Node next;

        Node(int d) {
            data = d;
            next = null;
        }
    }

    public static void main(String[] args) {
        LinkedList linkedlist = new LinkedList();

        // Assign values
        linkedlist.head = new Node(1);
        Node second = new Node(2);
        Node third = new Node(3);

        // Connect the nodes
        linkedlist.head.next = second;
        second.next = third;

        // Printing nodes
        while (linkedlist.head != null) {
            System.out.print(linkedlist.head.data + " ");
            linkedlist.head = linkedlist.head.next;
        }

    }
}
